src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Linux.cs

```diff
@@ -33,11 +33,13 @@ public static string[] GetPortNames()
                     // checking for that for non-ttyS entries is incorrect as some uart
                     // devices are incorrectly filtered out
                     bool isTtyS = entry.Name.StartsWith("ttyS");
+                    bool isTtyGS = entry.Name.StartsWith("ttyGS");
                     if ((isTtyS &&
                          (File.Exists(entry.FullName + "/device/id") ||
                           Directory.Exists(entry.FullName + "/device/of_node"))) ||
                         (!isTtyS && Directory.Exists(entry.FullName + "/device/tty")) ||
-                        Directory.Exists(sysUsbDir + entry.Name))
+                        Directory.Exists(sysUsbDir + entry.Name) ||
+                        isTtyGS)
```

Comment:
For `isTtys`, there are several additional checks being done when it's true.  No similar checks are needed when `isTtyGS` is true?

Reply:
As described in issue #1677, I can only say that the port structure does not necessarily behave like ttySx. As mentioned by @krwq, GigaSet phones also use this serial port and I have no idea what features their serial device has.
We could check if the folder "/sys/class/tty/ttyGS0/dev" exists. However, the fact that this folder exists on my system does not mean that it exists on other ports.
So I would suggest we keep this check.

Reply:
I generally prefer being more restrictive with checks and fix it when someone reports the problem but at the same time `ttyGS` seem to be documented to be related with serial ports. `ttyS` have a lot of stuff which you normally don't want to show up when enumerating ports so I don't really feel strongly either way